cores/esp8266/Esp.h

```diff
@@ -23,6 +23,48 @@
 
 #include <Arduino.h>
 
+#ifndef PUYA_SUPPORT
+  #define PUYA_SUPPORT 1
+#endif
+#ifndef PUYA_BUFFER_SIZE
+  // Good alternative for buffer size is: SPI_FLASH_SEC_SIZE (= 4k)
+  // Always use a multiple of flash page size (256 bytes)
+  #define PUYA_BUFFER_SIZE 256
+#endif
+
+// Vendor IDs taken from Flashrom project
+// https://review.coreboot.org/cgit/flashrom.git/tree/flashchips.h?h=1.0.x
+#define SPI_FLASH_VENDOR_ALLIANCE    0x52    /* Alliance Semiconductor */
```

Comment:
This is a great help, thank you! there are some known issues with the flash size detection for some vendors, and there was talk of building such a table.

Reply:
As an after-thought: maybe it would be better to have this as an enum with assigned values.

Reply:
Adding Unknown is a good idea. Please add it as the last/biggest value, e.g.: 0xFF.
Don't touch getVendorId() function, it's ok as it is.
I suggest
- calling the enum flashVendor or similar
- use enum class instead of classic enum (new-ish C++ feature), which forces the enum values to be contained in the class name in a similar way as a namespace (reduces chance of name clash). This also simplifies the enum values, i.e.: class enum flashVendor {AMD = 0x01, AMIC = 0x37,...};
- adding a `flashVendor getFlashVendorFromId(uint8_t id)` function which implements a switch, where all known types fall through to single case that does `return (flashVendor)id;` and a default case that does `return Unknown`

Reply:
OK, will do... tomorrow.
I was just about to test what I had, but this will take a bit more and it is already close to 1am.
The rest is already present, but not yet tested.

Reply:
By the way, should we also have a toString function for it? (will take quite some flash space)

And also I wondered, is there some consensus about how to handle internal strings (used for internal purposes, not to be translated ever) and display strings?
Tasmota uses defines for the display strings, so you can simply use another .h file for translated versions.

Reply:
No to toString(). These aren't meant for display, or even user info really, but rather to check conditions and make certain decisions in the internal core behavior. If a core user needs the string table, it can be implemented on his side.
About internal strings in general, strings smaller than... I think it was 4 bytes, are ok to be const char *, while strings longer than that should go to flash with PROGMEM or F() or whatever.
About translations, this has been brought up before. Impementation of complete and generic support for translation requires huge refactoring of current code. It also requires design of the specific approach, there are several possibilities. It can be done in time, but there are higher priorities at the moment, so for now the language for internal strings is English, and it can't be changed.